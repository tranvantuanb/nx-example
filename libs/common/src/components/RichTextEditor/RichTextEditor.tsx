import React, { memo } from 'react';
import { Editor, EditorState } from 'draft-js';
import { stateToHTML } from 'draft-js-export-html';

export interface RichTextEditorProps {
  onChange?: (value: EditorState) => void;
  value?: EditorState;
  placeholder?: string;
  height?: number;
}

export const getPlainText = (value: EditorState) => {
  return value?.getCurrentContent()?.getPlainText('\u0001') ?? '';
};

export const getHTMLContent = (value) => {
  return stateToHTML(value.getCurrentContent());
};

export const RichTextEditor = memo(
  ({ value, onChange = () => {}, ...rest }: RichTextEditorProps) => {
    return (
      <Editor
        editorState={value || EditorState.createEmpty()}
        onChange={onChange}
        {...rest}
      />
    );
  }
);
