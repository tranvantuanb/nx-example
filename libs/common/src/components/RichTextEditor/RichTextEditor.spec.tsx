import { render } from '../../test';

import { RichTextEditor, RichTextEditorProps } from './RichTextEditor';

jest.mock('draft-js/lib/generateRandomKey', () => {
  let deterministicKey = 0;
  return () => deterministicKey++;
});

describe('RichTextEditor', () => {
  it('should render successfully', () => {
    const defaultProps: RichTextEditorProps = {};
    const { container } = render(<RichTextEditor {...defaultProps} />);
    expect(container).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
});
