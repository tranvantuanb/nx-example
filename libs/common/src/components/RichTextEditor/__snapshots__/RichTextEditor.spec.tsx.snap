// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RichTextEditor should render successfully 1`] = `
<div>
  <div
    class="DraftEditor-root"
  >
    <div
      class="DraftEditor-editorContainer"
    >
      <div
        class="notranslate public-DraftEditor-content"
        contenteditable="true"
        role="textbox"
        spellcheck="false"
        style="outline: none; user-select: text; white-space: pre-wrap; word-wrap: break-word;"
      >
        <div
          data-contents="true"
        >
          <div
            class=""
            data-block="true"
            data-editor="1"
            data-offset-key="0-0-0"
          >
            <div
              class="public-DraftStyleDefault-block public-DraftStyleDefault-ltr"
              data-offset-key="0-0-0"
            >
              <span
                data-offset-key="0-0-0"
              >
                <br
                  data-text="true"
                />
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
